admin/src/js/controllers/targets-edit.js

```diff
@@ -28,6 +29,19 @@ angular.module('controllers').controller('TargetsEditCtrl',
         $scope.locales = _.map(settings.locales, _.clone);
         if ($stateParams.id) {
           $scope.target = _.findWhere(settings.tasks.targets.items, { id: $stateParams.id });
+          if (typeof $scope.target.name === 'undefined') {
+            $scope.target.name = [];
+            $scope.locales.map(locale => locale.code).forEach((code) => {
+              const translation = $translate.instant($scope.target.translation_key, null, 'no-interpolation', code, null);
+              const content  = translation === $scope.target.translation_key ? '' : translation;
+
+              const nameObj = {
+                locale: code,
+                content: content
+              };
+              $scope.target.name.push(nameObj);
```

Comment:
When the submit happens all the translations will be saved to the settings which is pointless because we just ignore them anyway. This was also happening in the `$scope.submit` function below. I think we want the `name` field to go away eventually, so store the locales in a separate scope variable, eg: `$scope.name`. Then modify the `$scope.submit` so it doesn't create the name if it doesn't already exist - we need to keep it for backwards compatibility.